### PR TITLE
First pass at upstart script

### PR DESCRIPTION
**Pull request by stevenschlansker**

### CODE REVIEW

File: manifests/broker/service.pp [hunk 1]
```diff
@@ -23,31 +23,57 @@
...
+        } ~>
+
+        exec { 'purge_rcd':
+          command     => 'update-rc.d kafka remove',
+          refreshonly => true
+        } ->
+
+        file { "${service_name}.service":
+          ensure  => file,
+          path    => "/etc/init/${service_name}",
```
Comment (travees) [inline manifests/broker/service.pp]: The upstart config needs a `.conf` suffix or else upstart doesn't recognize it as a service config

**Comment by stevenschlansker** [comment]
@SegFaultAX No, I cannot, as dev-environment is not buildable:

```
    virtualbox: W: Failed to fetch http://ppa.launchpad.net/brightbox/ruby-ng/ubuntu/dists/trusty/main/binary-i386/Packages  Hash Sum mismatch
```

**Comment by stevenschlansker** [comment]
Ran some local tests now, and it at least compiles the catalog and spits out some files that make sense.

File: templates/upstart.erb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+start on runlevel [2345]
+stop on [!12345]
+
+respawn
+respawn limit 5 60
+
+umask 007
+limit nofile 65536 65536
+limit stack 10240 10240
```
Comment (travees) [inline templates/upstart.erb]: Is stack supposed to be bytes and not kbytes? I know ulimit takes kbytes but I think upstart expects bytes. 

Something segfaults when I try to use this config to start Kafka. It works if I remove the `limit stack` line.

**Pull request merged**